src/hci/manufacture.cpp

```diff
@@ -302,6 +304,49 @@ class ManufactureStatsButton: public StatsButton
 		{
 			displayIfHighlight(xOffset, yOffset);
 		}
+		drawNextDroidRank(xOffset, yOffset);
+	}
+
+	// show a little icon on the bottom, indicating what rank next unit will be
+	// remember that when it's a commander, same rank will require twice experience
+	void drawNextDroidRank(int xOffset, int yOffset)
+	{
+		const auto factory = controller->getObjectAt(objectIndex);
+		if (!factory) 
+		{
+			return;
+		}
+		const auto player = factory->player;
+		const auto exp = getTopExperience(player);
+		unsigned int lvl = 0;
+		if (!factory->pFunctionality->factory.psSubject)
+		{
+			// not showing icon when nothing is being built
+			return;
+		}
+		if (factory->pFunctionality->factory.psSubject->droidType == DROID_COMMAND)
+		{
+			lvl = getDroidLevel(exp, player, commandBrainComponent);
+		}
+		else if (factory->pFunctionality->factory.psSubject->droidType == DROID_CONSTRUCT
+		|| factory->pFunctionality->factory.psSubject->droidType == DROID_CYBORG_CONSTRUCT
+		|| factory->pFunctionality->factory.psSubject->droidType == DROID_CYBORG_REPAIR
+		|| factory->pFunctionality->factory.psSubject->droidType == DROID_REPAIR
+		|| factory->pFunctionality->factory.psSubject->droidType == DROID_TRANSPORTER
+		|| factory->pFunctionality->factory.psSubject->droidType == DROID_SUPERTRANSPORTER)
+		{
+			lvl = 0;
+		}
+		else 
+		{
+			lvl = getDroidLevel(exp, player, nullBrainComponent);
+		}
+		const auto expgfx = getDroidRankGraphicFromLevel(lvl);
+		if (expgfx != UDWORD_MAX)
+		{
+   		// FIXME: use offsets relative to template positon, not hardcoded values ?
+   		iV_DrawImage(IntImages, (UWORD)expgfx, xOffset + 45, yOffset + 4);	
```

Comment:
I think you meant to move these to the right... there are whitespaces at the beginning of these lines.